### PR TITLE
Implement create and get incoming webhook endpoints for APIv4

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
* Some refactoring of code from api package into app package
* Added new store function for getting webhooks for the system
* Implement POST /hooks/incoming endpoint for APIv4
* Implement GET /hooks/incoming endpoint for APIv4

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (mattermost/mattermost-api-reference#93)
- [x] All new/modified APIs include changes to the drivers
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: app/webhook.go [hunk 1]
```diff
@@ -192,3 +192,42 @@ func CreateWebhookPost(userId, teamId, channelId, text, overrideUsername, overri
...
+	if result := <-Srv.Store.Webhook().GetIncomingByTeam(teamId, page*perPage, perPage); result.Err != nil {
+		return nil, result.Err
+	} else {
+		return result.Data.([]*model.IncomingWebhook), nil
+	}
+}
+
+func GetIncomingWebhooksPage(page, perPage int) ([]*model.IncomingWebhook, *model.AppError) {
+	if !utils.Cfg.ServiceSettings.EnableIncomingWebhooks {
+		return nil, model.NewAppError("GetIncomingWebhooksForTeamPage", "api.webhook.get_incoming.disabled.app_error", nil, "", http.StatusNotImplemented)
```
Comment (grundleborg) [inline app/webhook.go]: Looks like a copy-paste error here with "GetIncomingWebhooksForTeamPage".

File: store/sql_webhook_store.go [hunk 2]
```diff
@@ -183,15 +183,36 @@ func (s SqlWebhookStore) PermanentDeleteIncomingByUser(userId string) StoreChann
...
 	storeChannel := make(StoreChannel, 1)
 
 	go func() {
 		result := StoreResult{}
 
 		var webhooks []*model.IncomingWebhook
 
-		if _, err := s.GetReplica().Select(&webhooks, "SELECT * FROM IncomingWebhooks WHERE TeamId = :TeamId AND DeleteAt = 0", map[string]interface{}{"TeamId": teamId}); err != nil {
+		if _, err := s.GetReplica().Select(&webhooks, "SELECT * FROM IncomingWebhooks WHERE DeleteAt = 0 LIMIT :Limit OFFSET :Offset", map[string]interface{}{"Limit": limit, "Offset": offset}); err != nil {
+			result.Err = model.NewLocAppError("SqlWebhookStore.GetIncomingList", "store.sql_webhooks.get_incoming_by_user.app_error", nil, "err="+err.Error())
```
Comment (grundleborg) [inline store/sql_webhook_store.go]: Should this be `NewAppError`?

File: store/sql_webhook_store.go [hunk 3]
```diff
@@ -183,15 +183,36 @@ func (s SqlWebhookStore) PermanentDeleteIncomingByUser(userId string) StoreChann
...
+func (s SqlWebhookStore) GetIncomingByTeam(teamId string, offset, limit int) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+		result := StoreResult{}
+
+		var webhooks []*model.IncomingWebhook
+
+		if _, err := s.GetReplica().Select(&webhooks, "SELECT * FROM IncomingWebhooks WHERE TeamId = :TeamId AND DeleteAt = 0 LIMIT :Limit OFFSET :Offset", map[string]interface{}{"TeamId": teamId, "Limit": limit, "Offset": offset}); err != nil {
 			result.Err = model.NewLocAppError("SqlWebhookStore.GetIncomingByUser", "store.sql_webhooks.get_incoming_by_user.app_error", nil, "teamId="+teamId+", err="+err.Error())
```
Comment (grundleborg) [inline store/sql_webhook_store.go]: And maybe this one too?

**Comment by jwilander** [comment]
@grundleborg fixed

**Comment by enahum** [comment]
@jwilander this PR needs a rebase

**Pull request merged**